src/index.js

```diff
@@ -110,18 +110,15 @@ export default class Calendar extends React.Component {
     return this.rootEl.current;
   }
 
-  bindEventHandlers = (props, prevProps) => {
+  bindEventHandlers = (props) => {
     const eventHandlerNames = Object.keys(props).filter((key) => /on[A-Z][a-zA-Z]+/.test(key));
 
     eventHandlerNames.forEach((key) => {
       const eventName = key[2].toLowerCase() + key.slice(3);
-      // For <Calendar onFocus={condition ? onFocus1 : onFocus2} />
-      if (prevProps && prevProps[key] !== props[key]) {
-        this.calendarInst.off(eventName);
-      }
+      this.calendarInst.off(eventName);
       this.calendarInst.on(eventName, this.props[key]);
```

Comment:
@iamchanii 
You should use `props`.
```
this.calendarInst.on(eventName, props[key]);
```

The `props`(nextProps) is passed from `shouldComponentUpdate` and `this.props` will be the previous property.


```
<Calendar onFocus={condition ? onFocus1 : onFocus2}
```
If you need to bind a handler that you need dynamically, as in the code below, it will not work correctly the first time.